src/Symfony/Component/Security/Guard/Tests/Firewall/GuardAuthenticationListenerTest.php

```diff
@@ -184,6 +204,45 @@ public function testHandleCatchesAuthenticationException()
         $listener->handle($this->event);
     }
 
+    /**
+     * @legacy
+     */
+    public function testLegacyInterfaceAvoidSupportsCall()
+    {
+        $authenticatorA = $this->getMock('Symfony\Component\Security\Guard\GuardAuthenticatorInterface');
+        $providerKey = 'my_firewall3';
+
+        $authenticatorA
+            ->expects($this->never())
+            ->method('supports');
```

Comment:
Ah I see, the solution is to create a class that implements the legacy interface which changes some value in its `supports()` method and then check that the value was never changed (thus meaning that `supports()` wasn't executed).